bc-backend/src/main/java/org/jboss/da/bc/impl/BuildConfigurationGeneratorImpl.java

```diff
@@ -100,6 +103,12 @@ private void validate(ProjectHiearchy hiearchy) throws IllegalStateException {
         if (!project.isUseExistingBc() && project.getProjectId() == null)
             throw new IllegalStateException("Project id is null for " + project.getGav());
 
+        Matcher m = bcNamePattern.matcher(project.getName());
+        if (!m.matches())
+            throw new IllegalStateException(
```

Comment:
We don't catch this exception in rest startAnalyse endpoint where is this method used.